src/api/lastfm.ts

```diff
@@ -0,0 +1,54 @@
+import { default as axiosOriginal } from "axios";
+import rateLimit from "axios-rate-limit";
+import { RecentTracks, Track } from "models/ScrobblePage";
+import { User } from "models/User";
+
+const axios = rateLimit(axiosOriginal.create(), {
+  maxRPS: 10,
+});
+
+const lastFmAPI =
+  process.env.REACT_APP_LAST_FM_API || "ab50ffdf56122232fa11a39059f77a42";
+const lastFmCall =
+  "https://ws.audioscrobbler.com/2.0/?format=json&api_key=" + lastFmAPI;
+
+async function getData<T>(url: string): Promise<T> {
+  try {
+    const response = await axios.get<T>(url);
+    return response.data;
+  } catch (error) {
```

Comment:
Consider handling different error codes, and take into account potential errors due to potential infringement of lastfm terms of service, like exceeding reasonable use cap. https://www.last.fm/api/tos

Reply:
To be handled in another PR.